### PR TITLE
Test order now shows lab results

### PR DESCRIPTION
**Pull request by edrisse**

### CODE REVIEW

File: app/common/test/views/lab-request.html [hunk 1]
```diff
@@ -207,13 +207,15 @@ <h3 class="panel-title">{{'TESTS_LIST'| translate}}</h3>
                             <tr>
                                 <th st-sort="test">{{'TEST' | translate}}</th>
                                 <th st-sort="test">{{'CATEGORY' | translate}}</th>
+                                <th st-sort="test">{{'TEST_RESULTS' | translate}}</th>
```
Comment (ynurmahomed) [inline app/common/test/views/lab-request.html]: Translation missing on clinic module.

screenshot from 2018-04-26 16-05-25

File: app/common/test/controllers/labRequestController.js [hunk 2]
```diff
@@ -195,6 +195,28 @@
     function showTestOrderDetails(testOrder) {
       vm.testsOfSelectedRequest = testOrder.testOrderItems;
       vm.testOrderInDetail = testOrder;
+      testOrderResultService.getTestOrderResult(testOrder.uuid).then(function (testOrderResult) {
+        var itemResults = testOrderResult.items;
+        itemResults.forEach(function (itemResult) {
+          var currentItemResult = vm.testsOfSelectedRequest.find(function (testItem) {
```
Comment (ynurmahomed) [inline app/common/test/controllers/labRequestController.js]: I would suggest that the this logic be implemented as a single method call to a service. It's not required but from experience it gets really hard to refactor when we assign values to controller properties inside nested callbacks.

**Comment by edrisse** [comment]
done @ynurmahomed

File: app/common/test/services/testOrderResultService.js [hunk 3]
```diff
@@ -70,18 +71,43 @@
...
     }
 
+    function getTestOrderConsolidateResult(testRequestUuid, testOrderUuid) {
+      return getTestOrderResult(testRequestUuid)
+        .then(function (testRequest) {
+          var testOrders = testRequest.items;
+          var testOrderWithResult = testOrders.find(function (testOrder) { return testOrder.uuid === testOrderUuid; });
+          return conceptService.getConceptByTestOrder(testOrderWithResult.uuid, 'custom:(uuid,display,answers,datatype,units)')
+            .then(function (concept) {
+              return $q(function (resolve) {
```
Comment (ynurmahomed) [inline app/common/test/services/testOrderResultService.js]: @edrisse Its not really necessary to return a promise from ```then```, you can just return the result.

https://docs.angularjs.org/api/ng/service/$q#chaining-promises

**Pull request merged**